### DISCUSSION TITLE
Authenticate towards private nuget-feed using PAT from Azure Devops

### DISCUSSION THREAD

**Discussion by AronAtOrebroKommun** [datalust/seq-tickets#2171] [Help]
Hi!
I've tried googling and also searching in the discussions here but no luck, so I'm creating this post in hopes of getting some clarity.

I am in the process of creating a SEQ Plugin for in-house deployment that sends adaptive cards to teams when triggered by alerts. The plugin seems to be working locally. Our setup involves Azure DevOps Server (on-premises), and we have a pipeline that packages and pushes the NuGet package to our private feed.

I'm now trying to check that the plugin works in our test environment. Our (trial) SEQ instance is containerized using Docker, (relevant to this situation?). The thing that I'm having trouble with is connecting to our private feed from within SEQ. I've created a Personal Access Token (PAT) from Azure DevOps Server. Despite attempts, including the use of a PAT with full access rights, authentication remains unsuccessful.

Sadly I have not been able to locate any documentation or guidelines that clarify whether this configuration is possible through the SEQ administrative interface. Any help would be greatly appreciated. 😊

**Comment by liammclennan** [comment]
Hi @QuickGrayEV 

Have you tried adding a private feed via Settings > Feed? If so, you should see an error in Seq's log (in /Logs or Settings > Diagnostics > Download a Full Diagnostic Report)

**Reply by AronAtOrebroKommun** [reply]
Thanks for answering so quickly!
I did add it via the Settings > Feed yes 😄 

The error im getting is a 401:

`Unable to install package {PackageId} v{PackageVersion} from {FeedId}","@l":"Error","@x":"System.Net.Http.HttpRequestException: Response status code does not indicate success: 401 (Unauthorized)`


Ive tried setting the username to 'unused' or my AD username (with the PAT being the password)

I've also tried setting the PAT as the username, but all of these variants are giving me 401.

**Reply by liammclennan** [reply]
I don't know much about authenticating with Azure DevOps server and PATs. Have you tried creating a console app and seeing if you can get the authentication to work?

**Reply by AronAtOrebroKommun** [reply]
I have not tried it but my best guess is that it's not supported as of now in SEQ (and thats fine 😄)

I'll try solving this using some sort of service account that has access to the ADS nuget-feed instead

**Comment by MaceWindu** [comment]
@QuickGrayEV, could you tell me why you develop Teams plugin intead of using existing one (https://www.nuget.org/packages/Seq.App.Teams.AdaptiveCard)?

@nblumhardt I think it makes sense to improve discoverability of plugins in UI/documentation. It's saddens me a bit to see how many duplicate plugins created :)

https://www.nuget.org/packages?q=Tags%3A%22seq-app%22&frameworks=&tfms=&packagetype=&prerel=true&sortby=created-desc

E.g. search by Telegram shows 6 different plugins already...  https://www.nuget.org/packages?q=Tags%3A%22seq-app%22+telegram&includeComputedFrameworks=true&prerel=true&sortby=created-desc
